追加: アプリ起動時にスプラッシュスクリーンを表示

asaday opened the pull request:
## 概要
アプリケーション起動から画面表示までの待ち時間に、スプラッシュスクリーンを表示して起動状態を明示します。

## 背景
起動してから画面が表示されるまで時間がかかり、アプリが動いているのか分からず不安になるため、起動中であることを視覚的に伝えるスプラッシュスクリーンを追加しました。

## 変更内容

- **app.on('ready')直後にスプラッシュを表示**
  - 起動後すぐにスプラッシュを表示し、アプリが動作中であることを明示
  - Updaterウィンドウ表示中もスプラッシュが背景に表示
  
- **重い同期処理の後回し**
  - DevTools Extensions削除処理をバックグラウンド実行
  - ログファイル切り詰め処理をバックグラウンド実行
  - これらの処理が起動をブロックしないよう遅延実行

- **スプラッシュの終了タイミング**
  - メインウィンドウの did-finish-load イベント後、100ms遅延してスプラッシュを閉じる
  - Vueアプリの初期レンダリング完了を待ってからメインウィンドウを表示

## 効果
- 起動時に「アプリが動いているか分からない」問題を解消
- 起動中であることを明確に伝える視覚的フィードバック

- 起動体験の改善

## テスト
- 開発環境での起動確認
- 本番環境（Updater有効）での起動確認
- スプラッシュ→メインウィンドウの切り替えが自然

<img width="396" height="296" alt="Monosnap mogador 2025-11-19 14-14-37" src="https://github.com/user-attachments/assets/011b2647-371e-41e9-bf51-4e0f54af474b" />

koizuka commented:
https://developer.mozilla.org/ja/docs/Web/API/Window/setImmediate
>非推奨;: この機能は非推奨になりました。まだ対応しているブラウザーがあるかもしれませんが、すでに関連するウェブ標準から削除されているか、削除の手続き中であるか、互換性のためだけに残されている可能性があります。

koizuka commented:
方向性は良さそうなので、移動したコードの元の部分はコメントで残さないで消してしまってよさそう

koizuka commented on splash-window.js:
```diff
@@ -0,0 +1,105 @@
+
+  const version = process.env.NAIR_VERSION || '';
+
+  const splashHtml = `
```
htmlをインラインリテラルで書いてしまうのは編集性が良くない気がする

asaday replied:
あれ、テストが・・
別途ファイルにした方がよさそうなのですが、このjsファイルがwebpackの対象外のところなので依存無い方が分かりやすいかなという判断で一旦全部書いてます。

koizuka replied:
この際AIにwebpackビルド対象にしてもらえば...!

koizuka replied:
まあちょっとややこしそうだからissueに詰むとかでもいいかw

asaday replied:
はい、やりようはありそうですがsplashの目的からすると初期化云々に巻き込まれる前に早く出したい感じです

koizuka replied:
いや、webpackはupdater.jsでもやってるけど、メインアプリとは別のバンドルツリーでビルドすれば単一jsとして出力されるから実行時のコストはないよ

koizuka commented:
PRの説明に画面イメージがあるといいかも

asaday commented:
<img width="394" height="292" alt="Monosnap mogador 2025-11-14 18-05-24" src="https://github.com/user-attachments/assets/3fad7e20-ce77-4a9f-a5a6-9149fb6381ab" />

asaday committed:
追加: アプリ起動時にスプラッシュスクリーンを表示

- N Airロゴを表示するスプラッシュウィンドウを実装
- メインウィンドウのコンテンツ読み込み完了時に自動的にスプラッシュを閉じる
- E2Eテストでのウィンドウフォーカス処理を改善（100ms間隔リトライ）
- electron-builderの設定にsplash-window.jsを追加
- アプリ終了時のcookies.flushStore問題を修正

asaday committed:
開発: スプラッシュ画面の実装と起動パフォーマンスの最適化

- スプラッシュウィンドウをsplash/フォルダに分離（HTML/JS/CSS）
- 5つのドットアニメーションによるローディング表示
- 起動時の不要な遅延を削除（開発モード2秒待機を排除）
- DevToolsの開放を非ブロッキング化
- session.flushStorageDataによるシャットダウン処理の最適化
- E2Eテストの追加（スプラッシュウィンドウのライフサイクル検証）

asaday committed:
修正: アプリケーション終了時のレースコンディションを解決

- shutdownCompleteハンドラーでsetTimeout(..., 0)を使用してmainWindow.close()を遅延実行
- mainWindow.on('close')との競合を回避し、closed イベントが確実に発火するように修正
- 終了時のログを詳細化し、各処理ステップを追跡可能に
- flushLogBufferSync()関数を追加してapp.exit(0)前にログバッファを同期的にフラッシュ
- second-instanceでmainWindowのisDestroyed()チェックを追加

この修正により、パッケージバイナリでの終了処理が正常に完了し、
以前の3秒ハング問題が解消された。

asaday committed:
修正: シャットダウン時のInitShutdownSequenceをスキップして高速化

- InitShutdownSequenceは package ビルドで5秒タイムアウトするため削除
- N Air はクラッシュハンドラープロセスを使用していないため不要
- IPC.disconnect() でクリーンアップされるため明示的な呼び出しは不要
- シャットダウン時間が5.5秒から0.4秒に改善 (92%削減)
- @track デコレーターの戻り値を修正 (async 関数の Promise を正しく返す)

経緯:
package ビルドでアプリ終了に3秒以上かかる問題を調査。
mainWindow の close イベントでレースコンディションが発生していたため
setTimeout(..., 0) で修正したが、依然として5秒以上かかっていた。

詳細なタイミングログを追加して調査した結果、InitShutdownSequence が
5126ms かかっていることが判明。この関数は obs-studio-node の
Crashpad ハンドラープロセスに登録解除を通知するが、N Air では
Crashpad が起動していないため5秒タイムアウトするだけだった。

Streamlabs Desktop では OBS_API_initAPI() 時に Crashpad を起動し
named pipe で通信しているが、N Air にはその実装がない。
また、obs-studio-node の公式テストでも InitShutdownSequence を
呼んでおらず、IPC.disconnect() のみでクリーンアップされる。

asaday committed:
開発: シャットダウンシーケンスのログを整理とテストのデバッグログをコメントアウト

- app/services/app/app.ts: 詳細なタイミングログを削除し、開始・完了・エラーのみに簡素化
- try-catch-finally で shutdownComplete の確実な送信を保証
- test/helpers/modules/core.ts: focusWindow のデバッグログをコメントアウト

yusuke-shibakiri commented:
動作確認しました。
問題ないかなと思いつつも、特に急ぎでなければデザインに関してはデザイナー側で検討させていただければと思います。

taroc-dw commented on test/helpers/modules/core.ts:
```diff
@@ -133,17 +134,49 @@ export async function getFocusedWindowId(): Promise<string> {
 export async function focusWindow(winIdOrRegexp: string | RegExp): Promise<boolean> {
   const client = await getClient();
   const handles = await client.getWindowHandles();
+  //console.log(`focusWindow: Found ${handles.length} window handles`);
```
コメントアウトされているコードやconsole.logが結構残っているので不要なのであれば掃除しても良さそうです

koizuka replied:
確かに

taroc-dw reviewed (approved):
実装自体は問題ないと思うので ✅ しておきますー

asaday committed:
開発: コメントアウトされたconsole.logを削除

asaday committed:
Merge branch 'n-air_development' into feature/add-splash-screen

asaday committed:
Merge branch 'n-air_development' into feature/add-splash-screen
